src/utils/cli-auth.ts

```diff
@@ -48,9 +57,29 @@ export function parseCLIAuth(options: CLIAuthOptions): Partial<SynapseSetupConfi
   const privateKey = options.privateKey || process.env.PRIVATE_KEY
   const walletAddress = options.walletAddress || process.env.WALLET_ADDRESS
   const sessionKey = options.sessionKey || process.env.SESSION_KEY
-  const rpcUrl = options.rpcUrl || process.env.RPC_URL
   const warmStorageAddress = options.warmStorageAddress || process.env.WARM_STORAGE_ADDRESS
 
+  // Determine RPC URL with priority: explicit rpcUrl > RPC_URL env > network flag/env > default
+  let rpcUrl: string | undefined
+  if (options.rpcUrl || process.env.RPC_URL) {
+    // Explicit RPC URL takes highest priority
+    rpcUrl = options.rpcUrl || process.env.RPC_URL
+  } else {
+    // Try to use network flag/env var
+    const network = (options.network || process.env.NETWORK)?.toLowerCase().trim()
+    if (network) {
+      // Validate network value
+      if (network !== 'mainnet' && network !== 'calibration') {
+        throw new Error(`Invalid network: "${network}". Must be "mainnet" or "calibration"`)
+      }
+      // Convert network to RPC URL
+      rpcUrl = RPC_URLS[network as 'mainnet' | 'calibration']?.websocket
+      if (!rpcUrl) {
+        throw new Error(`RPC URL not available for network: "${network}"`)
+      }
+    }
+  }
```

Comment:
we could add to `src/core/utils` and re-use since we'll likely need this there as well